Server/db/mongo/modules/checkModule.js

```diff
@@ -22,7 +22,27 @@ const dateRangeLookup = {
 
 const createCheck = async (checkData) => {
   try {
+    const { monitorId, status } = checkData;
+
+    const n = await Check.countDocuments({ monitorId }) + 1;
+
     const check = await new Check({ ...checkData }).save();
+
+    const monitor = await Monitor.findById(monitorId);
+
+    if (!monitor) {
+      throw new Error("Monitor not found");
+    }
+
+    if (monitor.uptimePercentage === undefined) {
+      monitor.uptimePercentage = (status === true) ? 1 : 0;
+    } else {
+      monitor.uptimePercentage =
+        (monitor.uptimePercentage * (n - 1) + (status === true ? 1: 0)) / n;
+    }
+
+    await monitor.save();
```

Comment:
Nice work, this strategy worked just as we discussed!

Reply:
Quick maths! Thanks to you!